Join exisiting improvements

ndp-opendap opened the pull request:
Implemented dimension caching for joinExisiting. This allows the ncml_handler to cache the dimensions the aggregation member datasets during the first access (which may take a long time!) so that they can be retrieved very quickly for subsequent accesses. Tested against an aggregation of 121 HDF4 files (~330MB each) the time for the first retrieval of the aggregation DDS was 52 seconds. The time for the second retrieval was 0.15 seconds. Woot!

ndp-opendap committed:
Starting to hookup dimension caching prototype

ndp-opendap committed:
First crack at caching granule dimensions is working

ndp-opendap committed:
Reworked caching to use a separate disk file for each granule's (data file's) dimension cache. This way if the file is used in multiple aggregations it only needs a single cache, if the file get's dropped from the aggregation (ala FMRC) no harm is done, and if the source data for a single granule/file changes then only that one dimension cache file needs to be updated.

ndp-opendap committed:
Merged master to joinExisiting_improvements branch

ndp-opendap committed:
Small changes to dimension caching

ndp-opendap committed:
Refactored dimension caching to use a child of BESFileLockingCache for multiprocess safety. Works. woot.

ndp-opendap committed:
Refactored code to eliminate changes to the API

ndp-opendap committed:
Rewinding changes for BESFileLocking API in favor of less invasive use of the cache prefix as base for control file name.

mrpatrickwest commented:
I'm a huge fan of class comments and method comments, no matter if they are obvious or not. Just helps with future development, understand what the method is doing without having to look at the code, what and why there's a subclass, whatever.

I personally don't understand the benefit of using #if 0 in the code. If you don't need the code anymore then just delete it. If you want to look at it later or bring it back then take a look at a previous revision.

Otherwise changes make sense, follow-up on the changes made to caching earlier in BES.

ndp-opendap committed:
Comments, comments, comments

ndp-opendap committed:
Pruned unsed blocks of code.

ndp-opendap committed:
Pruned unsed blocks of code.

ndp-opendap commented:
I added comments to the new class' methods and header file and removed the #if 0 statements that where shutting off cruft that I wrote in the process of arriving at this solution. I left the #if 0 blocks  that were made by others.

jgallagher59701 commented:
Merge!

1+ comments

1+ '#if 0 ... ' until we're sure its correct to remove that, then delete it.

ndp-opendap committed:
Patched bes configuration prototypes, added some instrumentation.

ndp-opendap committed:
Patched bes configuration prototypes, added some instrumentation.

ndp-opendap committed:
Adding cache directory to test harness
